openedx/core/djangoapps/call_stack_manager/tests/tests.py

```diff
@@ -0,0 +1,162 @@
+        1. calls in decorated function should not get tracked
+        """
+        with LogCapture() as l:
+            donottrack_func()
+            self.assertEqual(len(l.records), 0, msg="Check @donottrack. should not log anything here!")
+
+    def test_parameterized_donottrack(self):
+        """ Test for parameterized @donottrack
+        1. Should not log calls of classes specified in the decorator @donotrack
+        """
+        with LogCapture() as l:
+            donottrack_parent_func()
+
+            # Example - logging new call stack for openedx.core.djangoapps.call_stack_manager.tests.models.modelmixin
+            latest_log = l.records[-1].getMessage()[:l.records[-1].getMessage().find(':')]
+            latest_class = latest_log[latest_log.rfind('.') + 1:]
+
+            # desired latest class here
+            desired_class = "ModelMixinCSM"
+
+            self.assertEqual(latest_class, desired_class,
+                             msg="The latest log should be of the class" + desired_class + "not" + latest_class)
+
+    def test_nested_parameterized_donottrack(self):
+        """ Tests parameterized nested @donottrack
+        1. should not track call of classes specified in decorated with scope bounded to the respective class
+        """
+        with LogCapture() as l:
+            #  class with CallStackManager as Manager
+            modelAnotherCSM_obj = ModelAnotherCSM(id_field=1, name_field="The Crownless again shall be king")
+            modelAnotherCSM_obj.save()
+
+            donottrack_parent_func()
+
+            # Example - logging new call stack for openedx.core.djangoapps.call_stack_manager.tests.models.modelmixin
+            latest_log = l.records[-1].getMessage()[:l.records[-1].getMessage().find(':')]
+            latest_class1 = latest_log[latest_log.rfind('.') + 1:]
+
+            latest_log = l.records[-2].getMessage()[:l.records[-2].getMessage().find(':')]
+            latest_class2 = latest_log[latest_log.rfind('.') + 1:]
+
+            actual_sequence = [latest_class2, latest_class1]
+            desired_sequence = ["ModelAnotherCSM", "ModelMixinCSM"]
+            self.assertEqual(actual_sequence, desired_sequence, msg=str(l))
+
+
+@donottrack('ModelAnotherCSM')
+def donottrack_child_func():
+    """ Function for decorator @donottrack
+    """
+    # should not be tracked
+    ModelAnotherCSM.objects.filter(id_field=1)
+
+    # should be tracked
+    ModelMixinCSM.objects.filter(id_field=1)
+
+
+@donottrack('ModelMixinCSM')
+def donottrack_parent_func():
+    """ Function for decorator @donottrack
+    """
+    # should not  be tracked
+    ModelMixinCSM.objects.filter(id_field=1)
+
+    # should be tracked
+    ModelAnotherCSM.objects.filter(id_field=1)
+
+    donottrack_child_func()
+
+
+@donottrack()
+def donottrack_func():
+    """ Function for decorator @donottrack
+    """
+    ModelMixin.objects.all()
+    donottrack_func_child()
+    ModelMixin.objects.filter(id_field=1)
+
+
+@donottrack()
```

Comment:
I think you should probably leave this `@donottrack` off, to test that functions that aren't explicitly un-tracked, but that are called within an un-tracked function, are still not tracked.